tests/catch/unit/device/hipGetSetDeviceFlags.cc

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <hip_test_common.hh>
+#include <iterator>
+#include <vector>
+#include <mutex>
+#include <condition_variable>
+/**
+ * Conformance test for checking functionality of
+ * hipError_t hipGetDeviceFlags(unsigned int* flags);
+ * hipError_t hipSetDeviceFlags(unsigned flags);
+ *
+ *
+ * hipGetDeviceFlags and hipSetDeviceFlags tests.
+ * Scenario1: Validates if hipGetDeviceFlags returns hipErrorInvalidValue for flags = nullptr.
+ * Scenario2: Validates if hipSetDeviceFlags returns hipErrorInvalidValue for invalid flags.
+ * Scenario3: Validates if flags returned by hipGetDeviceFlags is valid.
+ * Scenario4: Validates that flags set with hipGetDeviceFlags can be retrieved with
```

Comment:
"set with hipSetDeviceFlags"
Same typo in Scenario 5

Reply:
changed both